sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -125,8 +144,14 @@ trait SQLQueryTestHelper extends Logging {
     val df = session.sql(sql)
     val schema = df.schema.catalogString
     // Get answer, but also get rid of the #1234 expression ids that show up in explain plans
-    val answer = SQLExecution.withNewExecutionId(df.queryExecution, Some(sql)) {
-      hiveResultString(df.queryExecution.executedPlan).map(replaceNotIncludedMsg)
+    val answer = if (sql.toLowerCase().contains("as json")) {
```

Comment:
this is a bit tricky as it relies on string matching of SQL queries, how about we merging `replaceNotIncludedMsgJson` into `replaceNotIncludedMsg`?